### PR TITLE
Add narfkb Keyboard

### PR DESCRIPTION
**Pull request by dunk2k**
## Description

Added narfkb keyboard folder and child files

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* n/a

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/narfkb/halconf.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE. 
+*/
+
+#pragma once
+
```
Comment (waffle87) [inline keyboards/narfkb/halconf.h:25]: This file should be removed.

File: keyboards/narfkb/rules.mk [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+ENCODER_ENABLE = yes
+BACKLIGHT_ENABLE = no
+RGB_MATRIX_ENABLE = no
+RGBLIGHT_ENABLE = no
+MOUSEKEY_ENABLE = no
+CONSOLE_ENABLE = no
+COMMAND_ENABLE = no
+BOOTMAGIC_ENABLE = no
+LTO_ENABLE = no
```
Comment (waffle87) [inline keyboards/narfkb/rules.mk:11]: Unless you have a good reason for these 3 features to be disabled, it's generally preferred that they're enabled.
```suggestion
BOOTMAGIC_ENABLE = yes
EXTRAKEY_ENABLE = yes
MOUSEKEY_ENABLE = yes
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/narfkb/halconf.h (hunk 1 above)
Comment (dunk2k) [reply keyboards/narfkb/halconf.h]: File removed, in commit 3036ba7, as requested

**Review by waffle87 — APPROVED** [review]

File: keyboards/narfkb/info.json [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
+    },
+    "bootmagic": {
+        "matrix": [0, 0]
+    },
```
Comment (zvecr) [inline keyboards/narfkb/info.json:23]:
```suggestion
```

File: keyboards/narfkb/rules.mk [hunk 4]
```diff
@@ -0,0 +1,2 @@
+SERIAL_DRIVER = vendor
+ENCODER_ENABLE = yes
```
Comment (zvecr) [inline keyboards/narfkb/rules.mk:2]:
```suggestion
```

File: keyboards/narfkb/info.json [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    "processor": "RP2040",
+    "bootloader": "rp2040",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
```
Comment (zvecr) [inline keyboards/narfkb/info.json:19]:
```suggestion
        "nkro": true,
        "encoder": true
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/narfkb/info.json (hunk 5 above)
Comment (dunk2k) [reply keyboards/narfkb/info.json:19]: would the `#ifdef ENCODER_ENABLE` in narfkb.c still function if this change was implemented?

File: keyboards/narfkb/rules.mk (hunk 4 above)
Comment (dunk2k) [reply keyboards/narfkb/rules.mk:2]: would the `#ifdef ENCODER_ENABLE` in narfkb.c still function if this change was implemented?

File: keyboards/narfkb/info.json (hunk 5 above)
Comment (dunk2k) [reply keyboards/narfkb/info.json:19]: would the `#ifdef ENCODER_ENABLE` in narfkb.c still function if this change was implemented?

File: keyboards/narfkb/rules.mk (hunk 4 above)
Comment (zvecr) [reply keyboards/narfkb/rules.mk:2]: yes

**Review by zvecr — CHANGES_REQUESTED** [review]
Keyboard is missing a `readme.md` following the [template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md).

File: keyboards/narfkb/readme.md [hunk 6]
```diff
@@ -0,0 +1,2 @@
+# narfkb
```
Comment (zvecr) [inline keyboards/narfkb/readme.md:1]: You were linked a template to follow.

**Pull request closed without merging**